Build error in Launchpad

### Operating System

Ubuntu 24.04

### OpenRCT2 build

5fe3f7b97f167b1ad389cdaa5ff5680cc6073f40

### Base game

RollerCoaster Tycoon 2

### Area(s) with this issue?

_No response_

### Describe the issue

```
/usr/include/c++/13/bits/new_allocator.h:172: error: ‘operator delete’ called on unallocated object ‘<anonymous>’ [-Werror=free-nonheap-object]
  172 |         _GLIBCXX_OPERATOR_DELETE(_GLIBCXX_SIZED_DEALLOC(__p, __n));
      | 
/usr/src/openrct2-0.4.22-develop-0~2311-202505260018-g5fe3f7b97f~ubuntu24.04.1/src/openrct2-ui/windows/EditorScenarioOptions.cpp: In function ‘__merge_sort_with_buffer’:
/usr/src/openrct2-0.4.22-develop-0~2311-202505260018-g5fe3f7b97f~ubuntu24.04.1/src/openrct2-ui/windows/EditorScenarioOptions.cpp:2219:71: note: declared here
```

May have been caused by https://github.com/OpenRCT2/OpenRCT2/pull/24475 .

### Steps to reproduce

See above

### Attachments

_No response_

--- a/src/openrct2-ui/windows/EditorScenarioOptions.cpp
+++ b/src/openrct2-ui/windows/EditorScenarioOptions.cpp
@@ -2235,7 +2235,7 @@ namespace OpenRCT2::Ui::Windows
             }
 
             // Sort the rides alphabetically
-            std::stable_sort(_rideableRides.begin(), _rideableRides.end(), [](const RideId& a, const RideId& b) {
+            std::sort(_rideableRides.begin(), _rideableRides.end(), [](const RideId& a, const RideId& b) {
                 return String::compare(GetRideString(a), GetRideString(b), false) < 0;
             });
 

